lib/index-web.js

```diff
@@ -126,14 +126,18 @@ module.exports = function(config, auth, storage) {
   })
 
   // Search
-  app.get('/-/search/:anything', function(req, res, next) {
+  app.get('/-/search/:anything', can('access'), function(req, res, next) {
     var results = Search.query(req.params.anything)
     var packages = []
 
     var getData = function(i) {
       storage.get_package(results[i].ref, function(err, entry) {
         if (!err && entry) {
-          packages.push(entry.versions[entry['dist-tags'].latest])
+          auth.allow_access(entry.name, req.remote_user, function(err, allowed) { // TODO: This may cause performance issue?
```

Comment:
@juanpicado 
Request parameter not include 'package', so `can('access')` will not work. 

As you see, we search all matches in our storage, got a list of packages, push it into result if current user have permission to access it.

If you mean the comment. But i think it's not a problem before user have thousands of package in search result.

Any suggest?